### PR TITLE
Social | component

### PR DESCRIPTION
**Pull request by Yossitrx**
## Status 
**READY**

## Description
Social component 

## Todos
- [x] Tests

## Impacted Areas in Application
Bob-style

### CODE REVIEW

File: projects/ui-framework/src/lib/form-elements/social/social.component.scss [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  .social-input {
+    padding-right: 0
+  }
+
+  .icon-container mat-icon {
+    width: 24px !important;
+    height: 24px;
+    margin-right: 13px;
+  }
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/form-elements/social/social.component.scss]: remove sizing, size should come from iconSize enum
margin-right should be on the 12px or 14px

File: projects/ui-framework/src/lib/form-elements/social/social.component.scss [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  input {
+    color: $grey-600;
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/form-elements/social/social.component.scss]: why?

File: projects/ui-framework/src/lib/form-elements/social/social.component.scss [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  .prefix-container {
+    display: flex;
+    flex-direction: row;
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/form-elements/social/social.component.scss]: is this the input prefix? if so there is no need to add styling to it

File: projects/ui-framework/src/lib/form-elements/social/social.interface.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  Linkedin = 'linkedin',
+  Twitter = 'twitter'
+}
+
+export interface SocialInputConfig {
+  icon: Icons;
+  prefix: string;
+}
+
+
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/form-elements/social/social.interface.ts]: let's be consistent, have one empty line in between

File: projects/ui-framework/src/lib/form-elements/social/social.component.scss (hunk 2 above)
Comment (Yossitrx) [reply projects/ui-framework/src/lib/form-elements/social/social.component.scss]: design.

File: projects/ui-framework/src/lib/form-elements/social/social.component.scss (hunk 1 above)
Comment (Yossitrx) [reply projects/ui-framework/src/lib/form-elements/social/social.component.scss]: I will take it with the designers they implicitly said 24 on 24.

File: projects/ui-framework/src/lib/form-elements/social/social.interface.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+const twitter: SocialInputConfig = {
+  icon: Icons.twitter,
+  prefix: 'www.twitter.com/'
+};
+
+export const socialTypesConfig = {
+  facebook,
+  linkedin,
+  twitter
+};
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/form-elements/social/social.interface.ts]: i would move const to a separate file, it's for the use of the component only

File: projects/ui-framework/src/lib/form-elements/social/social.component.scss [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  display: inline-flex;
+  flex-direction: row;
+  width: 100%;
+  align-items: center;
+  b-input {
+    width: 100%;
+  }
+
+  .mat-form-field.has-prefix .mat-form-field-wrapper .mat-form-field-flex .mat-form-field-prefix {
+    padding-right: 0 !important;
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/form-elements/social/social.component.scss]: should work without the important i think

File: projects/ui-framework/src/lib/form-elements/social/social.component.scss (hunk 3 above)
Comment (Yossitrx) [reply projects/ui-framework/src/lib/form-elements/social/social.component.scss]: this is the div containing the prefix div.

File: projects/ui-framework/src/lib/form-elements/social/social.interface.ts (hunk 4 above)
Comment (Yossitrx) [reply projects/ui-framework/src/lib/form-elements/social/social.interface.ts]: will remove the extra space line

File: projects/ui-framework/src/lib/form-elements/social/social.component.html [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+         [errorMessage]="errorMessage"
+         [inputType]="inputTypes.text"
+         [label]="placeholder"
+         [value]="value"
+         [disabled]="disabled"
+         [required]="required"
+         [hintMessage]="hintMessage"
+         [hideLabelOnFocus]="true"
+         (inputEvents)="onInputEvents($event)">
+  <div class="prefix-container"input-prefix>
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/form-elements/social/social.component.html]: whats the wrapper for?
i think you can only use one element here

File: projects/ui-framework/src/lib/form-elements/social/social.component.scss (hunk 6 above)
Comment (Yossitrx) [reply projects/ui-framework/src/lib/form-elements/social/social.component.scss]: no. it was the last choice

File: projects/ui-framework/src/lib/form-elements/social/social.component.scss (hunk 1 above)
Comment (omrihecht) [reply projects/ui-framework/src/lib/form-elements/social/social.component.scss]: also naming can be icon, or remove the class and use b-icon here

File: projects/ui-framework/src/lib/form-elements/social/social.component.html (hunk 7 above)
Comment (Yossitrx) [reply projects/ui-framework/src/lib/form-elements/social/social.component.html]: I think you right here, will remove it.

File: projects/ui-framework/src/lib/form-elements/social/social.component.scss (hunk 1 above)
Comment (omrihecht) [reply projects/ui-framework/src/lib/form-elements/social/social.component.scss]: i think 24x24 is one of the sizes of icons we have, if not use the closest one, for sure not change icon size for a single component

**Comment by Yossitrx** [comment]
@omrihecht updated

File: projects/ui-framework/src/lib/form-elements/social/social.component.scss (hunk 2 above)
Comment (omrihecht) [reply projects/ui-framework/src/lib/form-elements/social/social.component.scss]: should be the same as other inputs, doesn't make sense if it's different for this one component

File: projects/ui-framework/src/lib/form-elements/social/social.component.scss (hunk 1 above)
Comment (Yossitrx) [reply projects/ui-framework/src/lib/form-elements/social/social.component.scss]: Removed it. it's our normal sizing

**Pull request merged**